internal/zinc-compile-core/src/test/scala/sbt/internal/inc/javac/JavaCompilerSpec.scala

```diff
@@ -95,7 +95,7 @@ class JavaCompilerSpec extends UnitSpec {
         val (result, _) = compile(local, Seq(input), Seq("-d", out.getAbsolutePath))
         result shouldBe true
         val clazzz = new URLClassLoader(Array(out.toURI.toURL)).loadClass("hasstaticfinal")
-        ClassToAPI(Seq(clazzz))
+        ClassToAPI(Seq(clazzz)).sortBy(_.definitionType.hashCode)
```

Comment:
`ClassToAPI` uses a `Set` to keep track of the APIs, then returns its result converted into a `Seq`.
Contraband generate the `hashCode()` method in a different way, depending on the class having lazy fields or not. Since all the lazy fields have been removed, it computes the hash by combining the hash of all the fields together instead of just calling `super.hashCode()` (See [JavaCodeGen.scala](https://github.com/sbt/contraband/blob/master/library/src/main/scala/sbt/contraband/JavaCodeGen.scala#L332-L345)). This should explain why we now have a random order.

Since we zip the `leftAPI` and `rightAPI` together, sorting the result seems to be the apropriate way to ensure we call `SameAPI` with the correct pairs.